### ISSUE TITLE
Mocking gevent makes impossible to use python-bayeux together with _posixsupprocess

### ISSUE THREAD

**Issue by edmondop** [SalesforceFoundation/python-bayeux#10]
This issue raise from my effort to create a custom task for CumulusCI to consume platform events from Salesforce in a test and assert they are correctly emitted.

I have created a simple task with the following import to test consumption / print events to screen:

`from python_bayeux import BayeuxClient`

This statement however trigger a monkey-patching of gvent https://github.com/SalesforceFoundation/python-bayeux/blob/master/src/python_bayeux/__init__.py#L32 which will make the _posixsubprocess module "None". As a consequence invoking my task fails after invoking sfdx

```

          File "/Users/edmondoporcu1/Library/Caches/pypoetry/virtualenvs/my-project-qe3xrZyf-py3.8/lib/python3.8/site-packages/cumulusci/core/config/SfdxOrgConfig.py", line 39, in sfdx_info
    p = sfdx("force:org:display --json", self.username, check_return=True)
  File "/Users/edmondoporcu1/Library/Caches/pypoetry/virtualenvs/my-project-qe3xrZyf-py3.8/lib/python3.8/site-packages/cumulusci/core/sfdx.py", line 50, in sfdx
    p.run()
  File "/Users/edmondoporcu1/Library/Caches/pypoetry/virtualenvs/my-project-qe3xrZyf-py3.8/lib/python3.8/site-packages/sarge/__init__.py", line 669, in run
    self.process = p = Popen(self.args, **self.kwargs)
  File "/Users/edmondoporcu1/.pyenv/versions/3.8.1/lib/python3.8/subprocess.py", line 855, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/edmondoporcu1/.pyenv/versions/3.8.1/lib/python3.8/subprocess.py", line 1638, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
AttributeError: 'NoneType' object has no attribute 'fork_exec'
```

What are the possible work-around to use python_bayeux in a library which also use subprocesses? Note that Cumulus uses _posixsubprocess as a part of sarge library

**Comment by adamlincoln** [comment]
Hi @edmondo1984 - thanks for the report.  This should be fixed in the 1.0.0 release, which is now on pypi.  As @halfdan pointed out in https://github.com/SalesforceFoundation/python-bayeux/pull/9 , this library should not call any of the gevent.monkey.patch_* methods itself.  It doesn't call patch_all() any longer, so you should be free to call patch_socket() and patch_ssl() in your own code to allow python-bayeux to work, but leave subprocess unpatched.

I'm fairly certain this will unblock you, so I'm going to close this issue.  But if it still doesn't work, please reopen or create a new issue.
